sympy/core/assumptions.py

```diff
@@ -173,7 +173,7 @@
     'zero           ->  even & finite',
 
     'prime          ->  integer & positive',
-    'composite      ==  integer & positive & !prime',
+    'composite      ->  integer & positive & !prime',
```

Comment:
I repeat, you should either buy "1 being composite" or "i.is_composite is None", not both.

Perhaps, I misunderstand something, but I can't guess what you are talking about, which hack...